src/Tribe/Repositories/Event.php

```diff
@@ -1781,14 +1781,19 @@ public function order_by( $order_by, $order = 'ASC' ) {
 	 */
 	public function filter_by_hidden_on_upcoming( $hidden ) {
 		$hidden = tribe_is_truthy( $hidden );
+		$hidden_posts = tribe( \Tribe\Events\Views\V2\Query\Hide_From_Upcoming_Controller::class )->get_hidden_post_ids();
```

Comment:
I am a lil worried about hitting some edge case limits on this. 

Any issues/concerns about grabbing all? Will this hit a wall at like 1000 hidden posts etc?

Reply:
Potentially, I don't think you are wrong that this could have an upper limit problem.

Wonder if we should have a filter that allows someone to disable this and move it back to the normal query.

Reply:
Yeah, I've hit several issues with similar queries. Sometimes queries get so big they will fail because of the size of the query text itself.

I guess considering time just doing a flag to use the old is an option, but it will not be easy for anyone to know when to use it.

Any thoughts on creating a temporary table to join on? Never done something like that in a WP world.

Alternatively we move this to a custom table to help optimize it, but that's a bigger project. Maybe this is the long term goal, a set of more optimizations in the custom table move.

Reply:
Other thing I wanted to point out, in the WP hooks somewhere someone may look at the `post__in` field on the query args array and iterate on some large list.